server/src/routes/api/v3/wallets/notify.js

```diff
@@ -0,0 +1,151 @@
+      const sentNetBalanceChange = tx.netBalanceChanges.find(netBalance => netBalance.address === totlePrimaryAddress)
+      const receivedNetBalanceChange = netBalanceChanges.find(netBalance => netBalance.address === watchedAddress)
+      console.log({ receivedNetBalanceChange })
+      const sentTx = createTxFromNetBalanceChange(tx, sentNetBalanceChange, { isSwap: true, externalId: 1 })
+      sentTx.from = watchedAddress
+      console.log({ sentTx })
+      const receivedTx = createTxFromNetBalanceChange(tx, receivedNetBalanceChange, { isSwap: true, externalId: 2 })
+      return [receivedTx, sentTx]
+    } else {
+      console.log(`processing the ${tx.hash} as erc20 transfer by looking into netBalanceChanges`)
+      return [createTxFromNetBalanceChange(tx, netBalanceChanges[0])]
+    }
+  } else if (tx.contractCall) {
+    console.log(`processing the ${tx.hash} as ERC20 token transfer`)
+    const { contractCall } = tx
+    const { params, contractType } = contractCall
+    if (contractType !== 'erc20') {
+      throw new Error(`unsupported contract type ${contractType} from tx ${tx.hash}`)
+    }
+    const { _to, _value } = params
+    return [{ ...tx, to: _to, value: _value, tokenAddress: tx.to }]
+  } else {
+    console.log(`Cannot process the ${tx.hash} transaction`)
+    return []
+  }
+}
+
+const updateWallet = async (tx, watchedAddress) => {
+  const userWallet = await UserWallet.findOne({ walletAddress: toChecksumAddress(watchedAddress) })
+  if (!userWallet) {
+    console.warn(`address ${watchedAddress} does not have a user wallet`)
+    return
+  }
+
+  let value
+  if (!(userWallet.balancesOnForeign.get(tx.tokenAddress))) {
+    value = tx.value
+  } else {
+    if (tx.to === watchedAddress) {
+      value = new BigNumber(userWallet.balancesOnForeign.get(tx.tokenAddress)).plus(tx.value)
+    } else {
+      value = new BigNumber(userWallet.balancesOnForeign.get(tx.tokenAddress)).minus(tx.value)
+    }
+  }
+  return UserWallet.updateOne({ walletAddress: toChecksumAddress(watchedAddress) }, { [`balancesOnForeign.${tx.tokenAddress}`]: value })
+}
+
+const createForeignWalletIfNeeded = async ({ watchedAddress, status }) => {
+  if (status !== 'confirmed') {
+    return
+  }
+  const walletAddress = toChecksumAddress(watchedAddress)
+  const userWallet = await UserWallet.findOne({ walletAddress })
+  const network = config.get('network.foreign.name')
+
+  if (userWallet.networks.includes(network) || (userWallet.networks.pendingNetworks && userWallet.networks.pendingNetworks.includes(network))) {
+    return
+  }
+
+  const pendingNetworks = [...userWallet.pendingNetworks, network]
+  await UserWallet.findOneAndUpdate({ walletAddress }, { pendingNetworks })
+
+  console.log(`starting a createForeignWallet job for ${JSON.stringify({ walletAddress: userWallet.walletAddress, network })}`)
+  // const job = await agenda.now('createForeignWallet', { userWallet, network: config.get('network.foreign.name') })
```

Comment:
I think we can stop creating foreign wallets on Ethereum for now, anyway the blocknative watch is not working. @LiorAgnin what you  think?

Reply:
@leonprou Yes, I agree.